addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -77,8 +77,14 @@ def _get_proxy_identification(self, company, proxy_type):
 
     def _cron_peppol_get_new_documents(self):
         # Retrieve all new Peppol documents for every edi user in the database
-        edi_users = self.env['account_edi_proxy_client.user'].search(
-            [('company_id.account_peppol_proxy_state', '=', 'active')])
+        # If `target_company_ids` is given in the context, retrieve the edi user
+        # from the given companies instead of all companies.
+        search_domain = [('company_id.account_peppol_proxy_state', '=', 'active')]
+
+        target_company_ids = self._context.get('target_company_ids')
```

Comment:
Add an optional parameter to pass the companies instead of the context

Reply:
Btw, since I have you here: when is it prefered to use a context value instead a function parameter ?

Reply:
@smetl 
In my opinion it would be better to refactor this function. I'm not sure it makes sense to parameterize this cron when we could easily just put the body of this function into it's own function called `_peppol_get_new_documents` , with line 203 onward
```for edi_user in edi_users``` `->` ```for edi_user in self```

we could write the cron as simply 
```python
def _cron_pepol_get_message_status(self):
    edi_users = ... all the active peppol users
    edi_users._peppol_get_new_documents
```

Reply:
I agree it would be nicer!